src/phoenix/server/api/helpers/experiment_run_filters.py

```diff
@@ -0,0 +1,796 @@
+
+    def create_experiment_runs_alias(self, experiment_id: ExperimentID) -> Any:
+        if self.get_experiment_runs_alias(experiment_id) is not None:
+            raise ValueError(f"Alias already exists for experiment ID: {experiment_id}")
+        experiment_index = self.get_experiment_index(experiment_id)
+        alias_name = f"experiment_runs_{experiment_index}"
+        aliased_table = aliased(models.ExperimentRun, name=alias_name)
+        self._aliased_experiment_runs[experiment_id] = aliased_table
+        return aliased_table
+
+    def get_experiment_runs_alias(self, experiment_id: ExperimentID) -> Any:
+        return self._aliased_experiment_runs.get(experiment_id)
+
+    def create_experiment_run_annotations_alias(self, experiment_id: ExperimentID) -> Any:
+        if self.get_experiment_run_annotations_alias(experiment_id) is not None:
+            raise ValueError(f"Alias already exists for experiment ID: {experiment_id}")
+        self._ensure_experiment_runs_alias_exists(
+            experiment_id
+        )  # experiment_runs are needed so we have something to join experiment_run_annotations to
+        experiment_index = self.get_experiment_index(experiment_id)
+        alias_name = f"experiment_run_annotations_{experiment_index}"
+        aliased_table = aliased(models.ExperimentRunAnnotation, name=alias_name)
+        self._aliased_experiment_run_annotations[experiment_id] = aliased_table
+        return aliased_table
+
+    def get_experiment_run_annotations_alias(self, experiment_id: ExperimentID) -> Any:
+        return self._aliased_experiment_run_annotations.get(experiment_id)
+
+    def get_experiment_index(self, experiment_id: ExperimentID) -> int:
+        return self._experiment_ids.index(experiment_id)
+
+    def _ensure_experiment_runs_alias_exists(self, experiment_id: ExperimentID) -> None:
+        if self.get_experiment_runs_alias(experiment_id) is None:
+            self.create_experiment_runs_alias(experiment_id)
+
+
+def _get_sqlalchemy_comparison_operator(
+    ast_operator: SupportedComparisonOperator,
+) -> Callable[[Any, Any], Any]:
+    if isinstance(ast_operator, ast.Eq):
+        return operators.eq
+    elif isinstance(ast_operator, ast.NotEq):
+        return operators.ne
+    elif isinstance(ast_operator, ast.Lt):
+        return operators.lt
+    elif isinstance(ast_operator, ast.LtE):
+        return operators.le
+    elif isinstance(ast_operator, ast.Gt):
+        return operators.gt
+    elif isinstance(ast_operator, ast.GtE):
+        return operators.ge
+    elif isinstance(ast_operator, ast.Is):
+        return operators.is_
+    elif isinstance(ast_operator, ast.IsNot):
+        return operators.is_not
+    elif isinstance(ast_operator, ast.In):
+        return lambda left, right: operators.contains_op(right, left)
+    elif isinstance(ast_operator, ast.NotIn):
+        return lambda left, right: operators.not_contains_op(right, left)
```

Comment:
Ideally, we should use [TextContains](https://github.com/Arize-ai/phoenix/blob/29a6d5a07a5fdd8d7c958d8d66f5af74dd70afaf/src/phoenix/trace/dsl/filter.py#L456), otherwise it turns into `LIKE` and special characters like `%` would need escaping.

Reply:
Good catch, I addressed with escaping since its smaller change to the current implementation. https://github.com/Arize-ai/phoenix/pull/5738/commits/10c5fd185e3b7c425c3bc61b83db83409a01e53e